documentation/src/main/doc/modules/kafka/pages/inbound.adoc

```diff
@@ -274,6 +274,31 @@ The `partitionkey` attribute is mapped to the record's key if any.
 
 include::consumer-rebalance-listener.adoc[]
 
+=== Receiving Kafka Records in Batches
+
+By default, incoming methods receive each Kafka record individually.
+Under the hood, Kafka consumer clients poll the broker constantly and receive records in batches, presented inside the `ConsumerRecords` container.
+
+In *batch* mode, your application can receive all the records returned by the consumer *poll*.
+
+To achieve this you need to set `mp.messaging.incoming.$channel.batch=true` *and* specify a compatible container type to receive all the data:
+
+[source, java, indent=0]
+----
+include::example$inbound/KafkaRecordBatchPayloadExample.java[tags=code]
+----
+
+The incoming method can also receive `Message<List<Payload>`, `KafkaBatchRecords<Payload>` `ConsumerRecords<Key, Payload>` types. These types give access :
+
+[source, java, indent=0]
+----
+include::example$inbound/KafkaRecordBatchExample.java[tags=code]
+----
+
+Note that the successful processing of the incoming batch record will commit the latest offsets for each partition received inside the batch. The configured commit strategy will be applied for these records only.
```

Comment:
"successful processing of the incoming batch record" -> "successful processing of the incoming record batch"?

Reply:
Also, what's "these records"? (I know it. The reader doesn't :-) )

Reply:
"these records" refer to the previously mentioned "latest offsets for each partition received inside the batch"